aws_embedded_metrics/serializers/log_serializer.py

```diff
@@ -31,6 +31,10 @@ def serialize(context: MetricsContext) -> List[str]:
         dimensions_properties: Dict[str, str] = {}
 
         for dimension_set in context.get_dimensions():
+
+            # Stringify numerical values and strip non-ascii characters from dimension set values
+            dimension_set = {k: str(v).encode('ascii', 'ignore').decode('ascii') for k, v in dimension_set.items()}
```

Comment:
Just a general question:

What if we stringified numerical values and stripped non-ascii characters when we call `put_dimensions()` and/or `set_dimensions()`, etc? Such as inside `validate_dimension_set()`

As opposed to iterating over all the dimensions when serializing?

Reply:
Iterating once per `put_dimensions()` call versus iterating over all dimensions when serializing should be the same. The latter requires less code.